Common/include/toolboxes/C2DContainer.hpp

```diff
@@ -90,18 +90,27 @@ class AccessorImpl
    * runtime internal value that depend on the number of rows/columns.
    * What values need setting depends on the specialization as not all have
    * members for e.g. number of rows and cols (static size optimization).
+   * \note Aligned dynamic allocation is disabled for compilation on MAC.
    */
-#define REAL_ALLOCATOR(EXTRA)                                           \
+#ifndef __APPLE__
```

Comment:
Addresses the problem reported by @economon on #753

Reply:
This now build on Mac, but I now see issues with the assert() calls on the first access of one of these containers at the start of the solver

Reply:
@economon I need more details, I do not have a Mac to go test this myself.

Reply:
Here's what I am seeing in Xcode (trace on left and assertion failure at bottom). Just let me know what else I can provide to help with this...

<img width="1679" alt="Screen Shot 2019-10-16 at 12 12 50 AM" src="https://user-images.githubusercontent.com/4896083/66896611-fe82b080-efa9-11e9-9728-9dc620f0e777.png">

Reply:
Humm, is this after you updated #790? (I think I recognise the function name) If so that probably means the data is not being allocated. Push your changes and I can have a look directly in #790.

Reply:
Do note that asserts are disabled when building with ninja unless `buildtype=debug`.
You may want to define NDEBUG for your optimized builds in Xcode as the cost of those checks is not negligible.

Reply:
Noted. I typically only build in serial debug mode in Xcode and keep a separate build at the command line in release mode w/ MPI, but I can add flags for this.

Btw, I also noticed that the new C2DContainer class has the assignment operator overloaded to perform a deep copy. Is it possible to do a shallow copy, i.e., just set the pointer of an existing initialized container to an existing one without allocating memory?

Reply:
With what is implemented it is not possible. If the objective is avoiding the double allocation of gradients, I would suggest using a VectorOfMatrix& that is set to primitive gradient when no allocation is needed, otherwise a different VectorOfMatrix object is resized and the the reference is set to that instead. This way there are no ownership problems (who frees the pointer and so on). References need an initializer but that can be either of the other containers.
If it's for a different purpose I can think of something.

Reply:
Aok, let me give that a try (yes it is for the gradients). #790 is looking good after merging everything, will be pushing the cleaned version shortly

Reply:
Looks like the choice of initializer does affect the solution, btw. Looks like the reconstruction gradient vector should use the auxiliary vector's initializer for correctness...